controllers/main.js

```diff
@@ -113,28 +111,43 @@ module.exports.renderPlay = async (req, res) => {
 module.exports.renderResults = async (req, res, next) => {
   const { id } = req.params;
   let playSessionData = await PlaySession.findOne({ sessionId: id });
-  const highscoreText = `${
-    remapDifficultyTexts[playSessionData.difficulty] + playSessionData.length
-  }`;
-  const difficultyFilter = `bestScores.${playSessionData.difficulty}`;
-  const highscoresData = JSON.stringify(
-    await User.find()
-      .sort({ [difficultyFilter]: -1 })
-      .limit(10)
-  );
 
   playSessionData = JSON.stringify(playSessionData);
   res.render("main/results", {
     extraStyles: resultsStyle + mediaButtonsStyle,
     playSessionData,
-    highscoresData,
-    highscoreText,
-    defaultProfileImg: playSettingsImage,
     sessionId: id,
   });
   next();
 };
 
+module.exports.fetchTopHighscores = async (req, res) => {
+  const { difficulty, length } = req.query;
+  const dbSearchFilter = `bestScores.${difficulty}.${length}`;
+  async function getTop10Users() {
+    const pipeline = [
```

Comment:
pipeline? what is this name

Reply:
Its a name that has been used in the examples i saw when looking up for a proper solution to the issue i had. Could change the variable name tho honestly not sure what kind of naming would be appropriate here.

Reply:
Basically what it does is – it is used as a filter that defines what kind of info we are looking when making a request to mongodb, and it filters out what data to send back to server from the database (it does not return the entire user object but only parts of it that are requested in this variable). I honestly have no clue what you would name a variable like that, I kept it as pipeline as per sample code because I thought it potentially was the naming convention for this kind of variable.

Reply:
Can you link where you found the sample code?

Reply:
Sorry, but I couldn't really find the place i got the sample from (firefox never saves browser history for whatever reason), but i tried searching for "mongodb pipeline" in google, and this is what i've found: https://studio3t.com/knowledge-base/articles/mongodb-aggregation-framework/

Reply:
Yes, debating the variable's naming and looking it up more helped it make more sense.